l10n_ch_payment_slip/models/payment_slip.py

```diff
@@ -443,18 +443,22 @@ def _get_scan_line_text_font(self, print_settings):
                         size=size)
 
     @api.model
-    def _get_address_lines(self, com_partner):
+    @tools.ormcache_context('self._uid', 'com_partner_id', keys=('lang',))
+    def _get_address_lines(self, com_partner_id):
         bvr_address_format = (
             self.env['ir.config_parameter'].get_param('bvr.address.format') or
             ADDR_FORMAT)
+        # clone the partner to not affect real partner
+        com_partner = self.env['res.partner'].browse(com_partner_id)
+        partner = self.env['res.partner'].new(com_partner.copy_data()[0])
         # use onchange to define our own temporary address format
         with self.env.do_in_onchange():
             # assign a fake country in case partner has no country set
-            com_partner.country_id = self.env['res.country'].new()
-            com_partner.country_id.address_format = bvr_address_format
-            address_lines = com_partner._display_address(
+            partner.country_id = self.env['res.country'].sudo().new({
```

Comment:
Do we really need the sudo ? does `new()` checks access rights?

Reply:
@yvaucher was in the old code. Was remove recently indeed. Anyway it does not hurt here and I prefer to stay safe and not break the whole thing on customers' face :smile: